pyproject.toml

```diff
@@ -0,0 +1,126 @@
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+name = "pdfminer.six"
+dynamic = ["version"]
+description = "PDF parser and analyzer"
+readme = "README.md"
+license = "MIT"
+requires-python = ">=3.8"
+authors = [
+    { name = "Yusuke Shinyama + Philippe Guglielmetti", email = "pdfminer@goulu.net" },
+]
+keywords = [
+    "layout analysis",
+    "pdf converter",
+    "pdf parser",
+    "text mining",
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Console",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Text Processing",
+]
+dependencies = [
+    "charset-normalizer >= 2.0.0",
+    "cryptography >= 36.0.0",
+]
+
+[project.optional-dependencies]
+dev = [
+    "atheris; python_version < '3.12'",
+    "black",
+    "mypy == 0.931",
+    "nox",
+    "pytest",
+]
+docs = [
+    "sphinx",
+    "sphinx-argparse",
+]
+image = [
+    "Pillow",
+]
+
+[project.scripts]
+"pdf2txt.py" = "tools.pdf2txt:main"
+"dumppdf.py" = "tools.dumppdf:main"
+
+[project.urls]
+Homepage = "https://github.com/pdfminer/pdfminer.six"
+
+[tool.hatch.version]
+source = "vcs"
+tag-pattern = '(\d\d\d\d\d\d\d\d)'
+
+[tool.hatch.build.targets.wheel]
+packages = ["pdfminer"]
```

Comment:
I like switching to pyproject.toml, because it is becoming the default in the python ecosystem afaik. But I'm not sure about hatch(ling). As far as I can understand it from my 10 min Google search, it is a tool that does many things, among which is publishing to pypi. Its functionality overlaps quite a bit with `uv` as well, and I'm thinking of moving to that in the future. 

What do you think of having this MR, but without hatch(ling)? 

And building with the python build package as described [here on the python packaging user guide](https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#checking-out-the-project-and-building-distributions)?

Reply:
Yes, we could quite easily just use plain old setuptools without any issues, there isn't anything particularly complicated in the build.  The Python build package is helpful as well.

Reply:
I can take a bit of time to revise the PR to just use setuptools in the next couple of days.